### PR TITLE
Adds minor protection to emag clicks

### PR DESCRIPTION
**Pull request by Poojawa**
:cl:
balance: Added sanity checks to emag usage
/:cl:

### CODE REVIEW

**Comment by deathride58** [comment]
This actually doesn't do anything but restrict potential emag effects to only specific hardcoded types. The code that actually reduces emag charges is in modular_citadel, which this PR doesn't touch

**Comment by Poojawa** [comment]
The only things this hardcodes against is mob/living that isn't a simple animal bot (for emagging beepsky, etc), drones, or silicons. You can't emag humans, after all. 

Turfs and storage containers that aren't lock boxes are also assured to not to. everything else fires off with emag_act, including charge use processing.

**Comment by deathride58** [comment]
https://github.com/Citadel-Station-13/Citadel-Station-13/blob/master/modular_citadel/code/game/objects/ids.dm#L85 This is where charge usage is handled. Your PR doesnt touch it

**Comment by Poojawa** [comment]
That's.. correct, yes. It's not intended to be changing that. This solves the reported problem of Emags incorrectly using charges when inserted into backpacks or used on turfs.

**Comment by deathride58** [comment]
Because this PR doesn't touch that code, this PR does not stop emags from consuming charges when the proc that the modular code overrides returns early.

**Comment by Poojawa** [comment]
```
 That's a cryptographic sequencer
It's a card with a magnetic strip attached to some circuitry.
It is a tiny item.
It has 10 charges left.
You put the cryptographic sequencer into the backpack.
 That's a cryptographic sequencer
It's a card with a magnetic strip attached to some circuitry.
It is a tiny item.
It has 10 charges left.
```

works just fine?

**Comment by Poojawa** [comment]
fixed.

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -95,7 +95,33 @@
...
+		playsound(src, 'sound/effects/light_flicker.ogg', 30, 1)	//Tiiiiiiny warning sound to let ya know your emag's almost dead
+
+	if(isturf(A))
+		return
+	if(istype(A,/obj/item/storage/lockbox))
+		A.emag_act(user)
+		uses = max(uses - 1, 0)
+		if(!uses)
+			user.visible_message("<span class='warning'>[src] fizzles and sparks. It seems like it's out of charges.</span>")
+			playsound(src, 'sound/effects/light_flicker.ogg', 100, 1)
```
Comment (deathride58) [inline code/game/objects/items/cards_ids.dm:113]: You can reduce the copypasta here by making use of `goto`

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -95,7 +95,33 @@
...
+	if(istype(A,/obj/item/storage))
+		return
+	if(!(isobj(A) || issilicon(A) || isbot(A) || isdrone(A)))
+		return
+	else
```
Comment (deathride58) [inline code/game/objects/items/cards_ids.dm:118]: This `else` here isn't necessary, as the check right above is an `if(); return`, and all of the other checks have a return at the end of them

**Review by deathride58 — APPROVED** [review]
It's spaghetti, but it'll work just fine, my own gripes about hardcoded typechecks aside

File: code/game/objects/items/cards_ids.dm (hunk 1 above)
Comment (Poojawa) [reply code/game/objects/items/cards_ids.dm:113]: I don't know how to use that.

**Comment by deathride58** [comment]
Remind me later and I'll do a small cleanup pass, my own personal gripes with this PR can be fixed later

**Pull request merged**